insecure/integration/functional/test/gossipsub/scoring/ihave_spam_test.go

```diff
@@ -0,0 +1,343 @@
+package scoring
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/stretchr/testify/require"
+	corrupt "github.com/yhassanzadeh13/go-libp2p-pubsub"
+
+	"github.com/onflow/flow-go/insecure/corruptlibp2p"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/irrecoverable"
+	"github.com/onflow/flow-go/module/mock"
+	"github.com/onflow/flow-go/network/channels"
+	"github.com/onflow/flow-go/network/p2p"
+	"github.com/onflow/flow-go/network/p2p/scoring"
+	p2ptest "github.com/onflow/flow-go/network/p2p/test"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+// TestGossipSubIHaveBrokenPromises_Below_Threshold tests that as long as the spammer stays below the ihave spam thresholds, it is not caught and
+// penalized by the victim node.
+// The thresholds are:
+// Maximum messages that include iHave per heartbeat is: 10 (gossipsub parameter).
+// Threshold for broken promises of iHave per heartbeat is: 10 (Flow-specific) parameter. It means that GossipSub samples one iHave id out of the
+// entire RPC and if that iHave id is not eventually delivered within 3 seconds (gossipsub parameter), then the promise is considered broken. We set
+// this threshold to 10 meaning that the first 10 broken promises are ignored. This is to allow for some network churn.
+// Also, per hearbeat (i.e., decay interval), the spammer is allowed to send at most 5000 ihave messages (gossip sub parameter) on aggregate, and
+// excess messages are dropped (without being counted as broken promises).
+func TestGossipSubIHaveBrokenPromises_Below_Threshold(t *testing.T) {
```

Comment:
can this be run with `t.Parallel()`? same question for all the other tests

Reply:
https://github.com/onflow/flow-go/pull/4566/commits/8fbf9a4d67cc87973da518ffddeceba4fb9d1eea

Reply:
https://github.com/onflow/flow-go/pull/4566/commits/9f2378054a24e0785479b2ceba24c991028f69ff